feature/security/gnsi/credentialz/tests/hiba_authentication/hiba_authentication_test.go

```diff
@@ -0,0 +1,690 @@
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/gnsi/credentialz"
+	tpb "github.com/openconfig/kne/proto/topo"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/binding"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	username                 = "testuser"
+	password                 = "i$V5^6IhD*tZ#eg1G@v3xdVZrQwj"
+	userPrivateKeyFilename   = "users/testuser"
+	userCertFilename         = "users/testuser-cert.pub"
+	dutPrivateKeyFilename    = "hosts/dut"
+	dutCertFilename          = "hosts/dut-cert.pub"
+	caPublicKeyFilename      = "ca.pub"
+	hostCertificateVersion   = "v1.0"
+	hostCertificateCreatedOn = 1705962293
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func createNativeRole(t testing.TB, dut *ondatra.DUTDevice, role string) {
+	t.Helper()
+	switch dut.Vendor() {
+	case ondatra.NOKIA:
+		roleData, err := json.Marshal([]any{
+			map[string]any{
+				"services": []string{"cli", "gnmi"},
+			},
+		})
+		if err != nil {
+			t.Fatalf("Error with json Marshal: %v", err)
+		}
+
+		userData, err := json.Marshal([]any{
+			map[string]any{
+				"password": password,
+				"role":     []string{role},
+			},
+		})
+		if err != nil {
+			t.Fatalf("Error with json Marshal: %v", err)
+		}
+
+		SetRequest := &gpb.SetRequest{
+			Prefix: &gpb.Path{
+				Origin: "native",
+			},
+			Replace: []*gpb.Update{
+				{
+					Path: &gpb.Path{
+						Elem: []*gpb.PathElem{
+							{Name: "system"},
+							{Name: "aaa"},
+							{Name: "authorization"},
+							{Name: "role", Key: map[string]string{"rolename": role}},
+						},
+					},
+					Val: &gpb.TypedValue{
+						Value: &gpb.TypedValue_JsonIetfVal{
+							JsonIetfVal: roleData,
+						},
+					},
+				},
+				{
+					Path: &gpb.Path{
+						Elem: []*gpb.PathElem{
+							{Name: "system"},
+							{Name: "aaa"},
+							{Name: "authentication"},
+							{Name: "user", Key: map[string]string{"username": username}},
+						},
+					},
+					Val: &gpb.TypedValue{
+						Value: &gpb.TypedValue_JsonIetfVal{
+							JsonIetfVal: userData,
+						},
+					},
+				},
+			},
+		}
+		gnmiClient := dut.RawAPIs().GNMI(t)
+		if _, err = gnmiClient.Set(context.Background(), SetRequest); err != nil {
+			t.Fatalf("Unexpected error configuring User: %v", err)
+		}
+	default:
+		t.Fatalf("Unsupported vendor %s for deviation 'deviation_native_users'", dut.Vendor())
+	}
+}
+
+func setupUser(t *testing.T, dut *ondatra.DUTDevice) {
+	auth := &oc.System_Aaa_Authentication{}
+	user := auth.GetOrCreateUser(username)
+
+	if deviations.SetNativeUser(dut) {
+		// probably all vendors need to handle this since the user should have a role attached to
+		// it allowing us to login via ssh/console/whatever
+		createNativeRole(t, dut, "credz-fp-test")
+	} else {
+		user.SetPassword(password)
```

Comment:
A password is not helpful for a hiba authentication test

Reply:
This should be removed as part of the aforementioned refactoring.

Reply:
Removed all password configs